### ISSUE TITLE
label leakage

### ISSUE THREAD

**Issue by zhandand** [PyGCL/PyGCL#61]
https://github.com/PyGCL/PyGCL/blob/3e6357bc845c90cbec5497e87e5981c48b66e693/examples/GRACE_SupCon.py#L75
In the supervised contrastive learning, the nodes with the same labels are collected as the postive pairs. So in my opinion, in the constrative pretraining stage, we can't utilize all the data, otherwise the labels are leaked in the tuning stage

**Comment by zhandand** [comment]
Sorry, my mistake. It seems that you have dealt with this problem
https://github.com/PyGCL/PyGCL/blob/3e6357bc845c90cbec5497e87e5981c48b66e693/GCL/models/sampler.py#L12

**Comment by wehos** [comment]
The authors' solution does not seem correct. From my perspective, it should be 

```
extra_pos_mask[~data.train_mask]=False
extra_pos_mask[:, ~data.train_mask]=False
```

instead of 
```
extra_pos_mask[~data.train_mask][:, ~data.train_mask] = False 
```

This seems to be a very severe label leakage! Can anyone confirm with me? @Linyxus 

> https://github.com/GraphCL/PyGCL/blob/33c7d1c53c6e0f5458122ecad59cd69f9e1d4af6/GCL/model/sampler.py#L124
> 
> 
> 
> https://github.com/GraphCL/PyGCL/blob/33c7d1c53c6e0f5458122ecad59cd69f9e1d4af6/GCL/model/sampler.py#L132
> 
> 
> 
> It seems that extra positive/negative will only be added for nodes in the training set.
> 
>  

 _Originally posted by @Linyxus in [#33](https://github.com/PyGCL/PyGCL/issues/33#issuecomment-1019492113)_
